exporter/metric/prometheus/sanitize.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: why is there a "internal/sanitize.go"? check if we can delete that.
+
+package prometheus
+
+import (
+	"strings"
+	"unicode"
+)
+
+const labelKeySizeLimit = 100
+
+// Sanitize returns a string that is trunacated to 100 characters if it's too
+// long, and replaces non-alphanumeric characters to underscores.
+func Sanitize(s string) string {
+	if len(s) == 0 {
+		return s
+	}
+	if len(s) > labelKeySizeLimit {
+		s = s[:labelKeySizeLimit]
+	}
+	s = strings.Map(sanitizeRune, s)
+	if unicode.IsDigit(rune(s[0])) {
+		s = "key_" + s
```

Comment:
... though I'd still like to be able to construct keys without checking for errors, maybe we should provide two options. If you construct a key w/ an error-free call (e.g., `key.New`, then it will silently emit a sanitized name. If you call a `Must()` variation, it will panic, perhaps there could also be a version that returns (core.Key, error). Not sure, but I'd like this kind of code to live somewhere else.

Reply:
+1 I like that. We took the same approach in tally. You can pass in a sanitizer when constructing the exporter. If a metric isnt sanitized, prometheus can throw an error.. So once could just swallow the errors and let them remain in the logs if they prefer..

Reply:
I'm going to resolve this and several remarks that would certainly belong in a separate PR.